### PR TITLE
feat: disable retries for conditionally idempotent file operations

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by shaffeeullah — APPROVED** [review]
Thanks , Denis!

File: src/file.ts [hunk 1]
```diff
@@ -3672,6 +3690,12 @@ class File extends ServiceObject<File> {
       typeof optionsOrCallback === 'function' ? optionsOrCallback : callback;
     const options =
       typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
+
+    // Do not retry if precondition option ifMetagenerationMatch is not set
```
Comment (tritone) [inline src/file.ts:3694]: As I said on the previous PR, I think user configurability should override this. @ddelgrosso1 and I will talk offline.

**Comment by ddelgrosso1** [comment]
After speaking with @tritone there is an additional option to be added to the retry configuration to bring this PR more in line with Go and Python. I am going to tag it do not merge while I add the flag.

**Review by shaffeeullah — APPROVED** [review]

**Review by tritone — APPROVED** [review]
Looks good, nice docs as well!

**Pull request merged**